pkg/envoy/sds/builder.go

```diff
@@ -0,0 +1,163 @@
+package sds
+
+import (
+	xds_core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	xds_auth "github.com/envoyproxy/go-control-plane/envoy/extensions/transport_sockets/tls/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/cache/types"
+	"github.com/openservicemesh/osm/pkg/certificate"
+	"github.com/openservicemesh/osm/pkg/envoy"
+	"github.com/openservicemesh/osm/pkg/envoy/secrets"
+	"github.com/openservicemesh/osm/pkg/errcode"
+	"github.com/openservicemesh/osm/pkg/identity"
+	"github.com/openservicemesh/osm/pkg/service"
+)
+
+type sdsBuilder struct {
+	proxy *envoy.Proxy
+
+	// Service certificate for this proxy
+	proxyCert *certificate.Certificate
+
+	// Requested certs to generate secrets for
+	requestedCerts []string
+
+	trustDomain string
+
+	serviceIdentitiesForServices map[service.MeshService][]identity.ServiceIdentity
+}
+
+func newSdsBuilder() *sdsBuilder {
+	return &sdsBuilder{}
+}
+
+func (b *sdsBuilder) SetProxy(proxy *envoy.Proxy) {
+	b.proxy = proxy
+}
+
+func (b *sdsBuilder) SetProxyCert(cert *certificate.Certificate) {
+	b.proxyCert = cert
+}
+
+func (b *sdsBuilder) SetRequestedCerts(requestedCerts []string) {
+	b.requestedCerts = requestedCerts
+}
+
+func (b *sdsBuilder) SetTrustDomain(trustDomain string) {
+	b.trustDomain = trustDomain
+}
+
+func (b *sdsBuilder) SetServiceIdentitiesForService(serviceIdentitiesForServices map[service.MeshService][]identity.ServiceIdentity) {
+	b.serviceIdentitiesForServices = serviceIdentitiesForServices
+}
+
+// Build generates SDS Secret Resources based on requested certs in the DiscoveryRequest
+func (b *sdsBuilder) Build() (sdsResources []types.Resource) {
```

Comment:
Let's drop the named return type since it doesn't improve readability here.
```suggestion
func (b *sdsBuilder) Build() []types.Resource {
```